db/sqldb/lrp_convergence.go

```diff
@@ -307,7 +307,7 @@ func (c *convergence) lrpsWithMetricTagChanges(ctx context.Context, logger lager
 			}
 		}
 
-		if !reflect.DeepEqual(desiredMetricTags, actualMetricTags) {
+		if actualMetricTags != nil && !reflect.DeepEqual(desiredMetricTags, actualMetricTags) {
```

Comment:
Wondering if metric tags can be empty? Looks like DesiredLRPUpdate has omitempty? So potentially they will be null if CC sends empty tags. Then later you won't be able to update them?

Reply:
We pushed a further change as discussed elsewhere to not allow Desired LRPs to be created without metric tags.

Thinking about the effect on clients of the BBS we think this is reasonable as for example the cloud controller code that constructs Desired LRPs has been [providing metric tags since January 2019](https://github.com/cloudfoundry/cloud_controller_ng/commit/105d5a7946354b623368f211d7d92033064b6a87).